testassets/FunctionalTestsWebsite/Infrastructure/InterceptorAttribute.cs

```diff
@@ -0,0 +1,40 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using Grpc.AspNetCore.Server;
+using Grpc.Core;
+using System;
+
+namespace FunctionalTestsWebsite.Infrastructure
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
```

Comment:
I can see class level attributes becoming a problem. It's essentially a service level option but has higher priority.

Reply:
Yes, that's a downside to multiple configuration options.

I don't have a problem with service level attribute configuration overriding service level code configuration. Attributes taking precedence in programming models is fairly common in .NET

Reply:
This is something we will need documentation for. I feel it could be a common pit of failure.